test/e2e/prometheus_test.go

```diff
@@ -250,69 +269,21 @@ spec:
 		f.AssertOperatorCondition(osConfigv1.OperatorDegraded, osConfigv1.ConditionFalse)
 		f.AssertOperatorCondition(osConfigv1.OperatorProgressing, osConfigv1.ConditionTrue)
 
-		t.Run(scenario.name, checkRemoteWrite(name, ctx))
-	}
-}
-
-func checkRemoteWrite(rwEndpointName string, ctx context.Context) func(*testing.T) {
-	return func(t *testing.T) {
-		remoteWriteCheckLogs(ctx, rwEndpointName, t)
-
-		remoteWriteCheckMetrics(ctx, t)
-	}
-}
-
-func remoteWriteCheckLogs(ctx context.Context, rwEndpointName string, t *testing.T) {
-	promLogs0, err := f.GetLogs(f.Ns, "prometheus-k8s-0", "prometheus")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	promLogs1, err := f.GetLogs(f.Ns, "prometheus-k8s-1", "prometheus")
-	if err != nil {
-		t.Fatal(err)
-	}
-	var promLogs strings.Builder
-	promLogErr := "prometheus logs are empty, expected to find log messages"
-	if i, _ := promLogs.WriteString(promLogs0); i == 0 {
-		t.Fatal(promLogErr)
-	}
-	if i, _ := promLogs.WriteString(promLogs1); i == 0 {
-		t.Fatal(promLogErr)
-	}
-
-	rwEndpointOpts := metav1.ListOptions{LabelSelector: labels.FormatLabels(map[string]string{"group": rwEndpointName})}
-
-	rwEndpointPodList, err := f.KubeClient.CoreV1().Pods(f.Ns).List(ctx, rwEndpointOpts)
-	if err != nil {
-		t.Fatal(err)
-	}
-	rwEndpointLogs, err := f.GetLogs(f.Ns, rwEndpointPodList.Items[0].ObjectMeta.Name, "")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if strings.Contains(promLogs.String(), `msg="Failed to send batch, retrying`) {
-		t.Fatal("unexpected prometheus log message, failed to send batch to remote write endpoint")
-	}
-	if strings.Contains(rwEndpointLogs, "remote error: tls: bad certificate") {
-		t.Fatal("remote write tls endpoint sees bad or no certificate")
+		t.Run(tc.name, func(t *testing.T) {
+			remoteWriteCheckMetrics(ctx, t, prometheusReceiveClient, tc.expected)
+		})
 	}
 }
 
-func remoteWriteCheckMetrics(ctx context.Context, t *testing.T) {
-	time.Sleep(1 * time.Minute)
-	for _, pod := range []string{
-		"prometheus-k8s-0",
-		"prometheus-k8s-1",
-	} {
-		f.ThanosQuerierClient.WaitForQueryReturn(
-			t, 1*time.Minute, fmt.Sprintf(`ceil(delta(prometheus_remote_storage_samples_pending{pod="%s"}[1m]))`, pod),
+func remoteWriteCheckMetrics(ctx context.Context, t *testing.T, promClient *framework.PrometheusClient, tests []remoteWriteTest) {
+	time.Sleep(2 * time.Minute)
```

Comment:
Why do we have a sleep here?

Reply:
The `sleep` here give the in-cluster Prometheus a chance to send some samples to the `remote_write` target. We don't want to fail by querying the target immediately when no samples have been send yet.

Reply:
I'll add a comment to that effect.

Reply:
we could also extend the timeout passed to `promClient.WaitForQueryReturn` to avoid hard-waiting for 2 minutes?